cli/cmd/upgrade.go

```diff
@@ -259,6 +261,7 @@ func (options *upgradeOptions) validateAndBuild(stage string, k kubernetes.Inter
 		return nil, nil, fmt.Errorf("could not build install configuration: %s", err)
 	}
 	values.Identity = identity
+	values.InstallNamespace = !identity.Issuer.External
```

Comment:
Why is this necessary? My understanding of our earlier conversation around omitting the persistence of the `--identity-external-issuer` flag is that its value isn't needed during upgrade. If during upgrade, the code needs to make decisions based on whether an issuer is external, then I don't think what the `isExternal()` function below is doing is reliable.

Reply:
>@ihcsim I see. I think @adleong had some very good points around the lack of need for persisting that as we can deduce even during upgrade whether cert-manager was used by simply looking at the contents of the secret.

Well if this is not set, then an attempt to include the namespace in the produced output will be made, right? Just like I noted here, this can be deduced without persisting. If thats unreliable, lets persist it then. The `isExternal()` function is doing pretty much the same thing that the identity service is doing when looking at the contents of the secret. 

I dont have the full context around the potential use and edge cases here, so its quite difficult for me to decide for sure. But why would it be unreliable? If the secret contains contents from cert manager and the flag is set to not use external issuer, it will not work at all after all. Again, up to you what we do here.

Reply:
> Well if this is not set, then an attempt to include the namespace in the produced output will be made, right? 

My thought is that `kubectl apply` will leave it alone, unless there are some differences. But yes, if the users created their own `linkerd` namespace, then `linkerd upgrade` should leave it alone, like what you did here.

> lets persist it then.

Sure. Thanks. Sorry for the back-and-forth.